pkg/openstack/instances.go

```diff
@@ -677,26 +728,86 @@ func nodeAddresses(srv *servers.Server, interfaces []attachinterfaces.Interface,
 	return addrs, nil
 }
 
-func getAddressesByName(client *gophercloud.ServiceClient, name types.NodeName, networkingOpts NetworkingOpts) ([]v1.NodeAddress, error) {
-	srv, err := getServerByName(client, name)
+func getAddressesByName(compute *gophercloud.ServiceClient, name types.NodeName, networkingOpts NetworkingOpts, network *gophercloud.ServiceClient) ([]v1.NodeAddress, error) {
+	srv, err := getServerByName(compute, name)
 	if err != nil {
 		return nil, err
 	}
 
-	interfaces, err := getAttachedInterfacesByID(client, srv.ID)
+	interfaces, err := getAttachedInterfacesByID(compute, srv.ID, network)
 	if err != nil {
 		return nil, err
 	}
 
 	return nodeAddresses(&srv.Server, interfaces, networkingOpts)
 }
 
+// getSubInterfaces returns the sub-interfaces attached via trunks to the specified node interfaces
+func getSubInterfaces(interfaces []attachinterfaces.Interface, network *gophercloud.ServiceClient) ([]attachinterfaces.Interface, error) {
+	klog.V(5).Infof("Node has %d interfaces '%v'", len(interfaces), interfaces)
+
+	var subports []attachinterfaces.Interface
+	for _, iface := range interfaces {
+		// Check if trunk ports are attached
+		listOpts := trunks.ListOpts{
+			PortID: iface.PortID,
+		}
+		allPages, err := trunks.List(network, listOpts).AllPages()
+		if err != nil {
+			klog.Errorf("Failed to list trunks: %v", err)
+			return subports, nil
+		}
+		allTrunks, err := trunks.ExtractTrunks(allPages)
+		if err != nil {
+			klog.Errorf("Failed to extract trunks: %v", err)
+			return subports, err
+		}
+		// Get subports attached to the trunks
+		for _, trunk := range allTrunks {
```

Comment:
Would it be easier to use trunk_details instead?

```
{
  "port": {
    <snip>
    "id": "713cc70f-fb49-4f7a-ae33-efb9eab93cb3",
    <snip>
    "trunk_details": {
      "sub_ports": [
        {
          "mac_address": "fa:16:3e:5b:bf:e1",
          "port_id": "0cd81ae6-df5d-47f6-ae14-a7ba706f4f23",
          "segmentation_id": 123,
          "segmentation_type": "vlan"
        }
      ],
      "trunk_id": "c264b0a2-b7d6-49a4-82b7-d1b42862d065"
    },
    <snip>
  }
}
```

At a glance Gophercloud doesn't support it, but we can still implement that support there if so. @pierreprinetti, do you find it feasible? How do we implement optional extension fields for Ports in Gophercloud? I feel like it should be possible by extending the Port structure we use?

Reply:
I thought about this, but I think it's the same number of API calls? We need to fetch the subports anyway in order to get the network name, so whether we get them from the trunk or the parent port doesn't make any difference, I think.

Reply:
Yeah, I think so, we end up with the same number of calls.